### PR TITLE
NAS-106553 / 21.06 / Add ability to set a different retention time for replicated snapshots

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
@sonicaj can you please review this?

**Comment by themylogin** [comment]
@sonicaj ping

File: src/middlewared/middlewared/plugins/zettarepl.py [hunk 1]
```diff
@@ -709,6 +709,14 @@ async def _replication_task_definition(self, pools, replication_task):
         if replication_task["lifetime_value"] is not None and replication_task["lifetime_unit"] is not None:
             definition["lifetime"] = lifetime_iso8601(replication_task["lifetime_value"],
                                                       replication_task["lifetime_unit"])
+        if replication_task["lifetimes"]:
+            definition["lifetimes"] = {
+                f"lifetime_{i}": {
+                    "schedule": zettarepl_schedule(lifetime["schedule"]),
+                    "lifetime": lifetime_iso8601(lifetime["lifetime_value"], lifetime["lifetime_unit"]),
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/zettarepl.py:716]: Will `lifetime_value` and `lifetime_unit` always exist ?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/zettarepl.py:716]: You are right, I forgot `required` in the `Dict`, thank you!

**Review by sonicaj — APPROVED** [review]

**Pull request merged**